internal/resources/managementclusterregistration/management_cluster_registration_test.go

```diff
@@ -0,0 +1,180 @@
+//go:build managementclusterregistration
+// +build managementclusterregistration
+
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package managementclusterregistration
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/pkg/errors"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/authctx"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/proxy"
+	managementclusterregistrationclient "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/managementclusterregistration"
+	testhelper "github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/testing"
+)
+
+func TestAcceptanceForManagementClusterRegistrationResource(t *testing.T) {
+	var provider = initTestProvider(t)
+
+	tKGsresourceName := fmt.Sprintf("%s.%s", "tanzu-mission-control_management_cluster", "test_tkgs")
+	tkgsName := acctest.RandomWithPrefix("a-tf-tkgs-test")
+
+	tKGmResourceName := fmt.Sprintf("%s.%s", "tanzu-mission-control_management_cluster", "test_tkgm")
+	tkgmName := acctest.RandomWithPrefix("a-tf-tkgm-test")
+
+	kubeconfigPath := os.Getenv("KUBECONFIG")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          testhelper.TestPreCheck(t),
+		ProviderFactories: testhelper.GetTestProviderFactories(provider),
+		CheckDestroy:      nil,
+		Steps: []resource.TestStep{
+			{
+				Config: getSimpleTKGsResourceWithDataSource(tkgsName),
+				Check: resource.ComposeTestCheckFunc(
+					checkResourceAttributes(provider, tKGsresourceName, tkgsName),
+				),
+			},
+			{
+				Config: getSimpleTKGmResourceWithDataSource(tkgmName),
+				Check: resource.ComposeTestCheckFunc(
+					checkResourceAttributes(provider, tKGmResourceName, tkgmName),
+				),
+			},
+			{
+				PreConfig: func() {
+					if kubeconfigPath == "" {
+						t.Skip("KUBECONFIG env var is not set for management cluster registration acceptance test")
+					}
+				},
+				Config: getSimpleTKGmResourceWithDataSourceWithKubeConfig(tkgmName, kubeconfigPath),
+				Check: resource.ComposeTestCheckFunc(
+					checkResourceAttributes(provider, tKGmResourceName, tkgmName),
+				),
+			},
+		},
+	},
+	)
+	t.Log("management cluster registration resource acceptance test complete!")
+}
+
+func getSimpleTKGsResourceWithDataSource(name string) string {
```

Comment:
Instead of simple can we name it as getTKGsResourceWithoutKubeconfigWithDataSource? same comment at other places as well.

Reply:
Done.